packages/core/src/extensions/lens-main-extension.ts

```diff
@@ -3,19 +3,49 @@
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
 
-import { LensExtension, lensExtensionDependencies } from "./lens-extension";
+import type { LensExtensionDependencies } from "./lens-extension";
+import { LensExtension } from "./lens-extension";
 import type { CatalogEntity } from "../common/catalog";
 import type { IComputedValue, IObservableArray } from "mobx";
 import { isObservableArray } from "mobx";
 import type { MenuRegistration } from "../features/application-menu/main/menu-registration";
 import type { TrayMenuRegistration } from "../main/tray/tray-menu-registration";
 import type { ShellEnvModifier } from "../main/shell-session/shell-env-modifier/shell-env-modifier-registration";
-import type { LensMainExtensionDependencies } from "./lens-extension-set-dependencies";
+import { Environments, getEnvironmentSpecificLegacyGlobalDiForExtensionApi } from "./as-legacy-globals-for-extension-api/legacy-global-di-for-extension-api";
+import type { InstalledExtension } from "./common-api";
+import type { CatalogEntityRegistry } from "../main/catalog";
+import type { NavigateForExtension } from "../main/start-main-application/lens-window/navigate-for-extension.injectable";
+import catalogEntityRegistryInjectable from "../main/catalog/entity-registry.injectable";
+import fileSystemProvisionerStoreInjectable from "./extension-loader/file-system-provisioner-store/file-system-provisioner-store.injectable";
+import loggerInjectable from "../common/logger.injectable";
+import navigateForExtensionInjectable from "../main/start-main-application/lens-window/navigate-for-extension.injectable";
 
-export class LensMainExtension extends LensExtension<LensMainExtensionDependencies> {
+interface LensMainExtensionDependencies extends LensExtensionDependencies {
+  readonly entityRegistry: CatalogEntityRegistry;
+  readonly navigate: NavigateForExtension;
+}
+
+export class LensMainExtension extends LensExtension {
   appMenus: MenuRegistration[] | IComputedValue<MenuRegistration[]> = [];
   trayMenus: TrayMenuRegistration[] | IComputedValue<TrayMenuRegistration[]> = [];
 
+  /**
+   * @ignore
+   */
+  declare readonly dependencies: LensMainExtensionDependencies;
```

Comment:
The `@ignore` is there so that we don't generate documentation for that field.

The declare is there because it will be assigned in the base class only.